### PR TITLE
Lambda Policy now has a class to manage View, and is more easily testable

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [x] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

The code that gets a Lambda Policy held all of its logic in one function. This made it hard to test, and to work with the corresponding view. This change breaks things out, making it better to work with the view, and makes the code testable.

This is a concept that could be used for the functionality that gets the function configuration as well.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/lambda/lambdaPolicy.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            'html',
+            localize(
+                'AWS.lambda.policy.title',
+                'Lambda Policy: {0}',
+                this._lambdaPolicyProvider.functionName
+            ),
+            -1
+        )
+
+        view.onDidDispose(() => {
```
Comment (mpiroc) [inline src/lambda/lambdaPolicy.ts]: Do we also need to unregister the `onDidDispose` handler, or does that happen automatically?

File: src/lambda/templates/lambdaTemplates.ts [hunk 2]
```diff
@@ -75,3 +56,31 @@ export class LambdaTemplates {
     <p>Version: <%= Version %>
     `
 }
+
+export class LambdaPolicyTemplates {
```
[the remark this answers is not in the document]
Comment (awschristou) [reply src/lambda/templates/lambdaTemplates.ts]: Cool I didn't know about that

File: src/test/lambda/lambdaPolicy.test.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    getLambdaPolicy: async () => {
+                        assert.equal(autoDisposeView!.getStatus(), LambdaPolicyViewStatus.Loading)
+                        resolve()
+
+                        return {}
+                    }
+                }
+            )
+
+            await autoDisposeView.load()
```
[the remark this answers is not in the document]
Comment (awschristou) [reply src/test/lambda/lambdaPolicy.test.ts]: Thanks - much cleaner

File: src/test/lambda/lambdaPolicy.test.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        await autoDisposeView.load()
+
+        assert.equal(autoDisposeView.getStatus(), LambdaPolicyViewStatus.Loaded)
+    })
+
+    it('enters error state', async () => {
+        autoDisposeView = new LambdaPolicyView(
+            {
+                functionName: 'function1',
+                getLambdaPolicy: async () => {
```
[the remark this answers is not in the document]
Comment (awschristou) [reply src/test/lambda/lambdaPolicy.test.ts]: It wouldn't take, I'll leave it

File: src/lambda/lambdaPolicy.ts (hunk 1 above)
Comment (awschristou) [reply src/lambda/lambdaPolicy.ts]: Good catch!

"Calls to subscribe return a Disposable which removes the event listener upon dispose."
https://code.visualstudio.com/docs/extensionAPI/patterns-and-principles#_events

**Pull request merged**